lisp/tree-sitter-debug.el

```diff
@@ -18,13 +18,60 @@
 (defvar-local tree-sitter-debug--tree-buffer nil
   "Buffer used to display the syntax tree of this buffer.")
 
+(defvar-local tree-sitter-debug--source-code-buffer nil
+  "Buffer of the source code tree-sitter is displaying.")
+
+(defgroup tree-sitter-debug nil
+  "Tree sitter debug and display features."
+  :group 'tree-sitter)
+
+(defcustom tree-sitter-debug-jump-buttons nil
+  "Wither to enable jump to node buttons in the tree-sitter-debug view.
+This can have a performace penalty on large buffers."
+  :type 'boolean
+  :group 'tree-sitter-debug)
+
+(defcustom tree-sitter-debug-highlight-jump-region nil
+  "Wither to highlight the node jumped to.
+Only takes effect if `tree-sitter-debug-jump-buttons' is non-nil."
+  :type 'boolean
+  :group 'tree-sitter-debug)
+
+
+(defun tree-sitter-debug--button-node-lookup (button)
+  "The function called when BUTTON was activated on a tree-sitter debug buffer."
+  (unless tree-sitter-debug--source-code-buffer
+    (error "No source code buffer set"))
+  (unless (buffer-live-p tree-sitter-debug--source-code-buffer)
+    (user-error "Source code buffer has been killed"))
+  (unless button
+    (user-error "This function must be called on a button"))
+  (let ((pos (button-get button 'points-to)))
+    (tree-sitter-debug--goto-node tree-sitter-debug--source-code-buffer
+                                  (car pos) (cdr pos))))
+
+(defun tree-sitter-debug--goto-node (buffer start end)
+  "Switch to BUFFER, centering on the region defined by START and END."
+  (switch-to-buffer-other-window buffer)
+  (goto-char start)
+  (if end
+      (push-mark end t tree-sitter-debug-highlight-jump-region)
```

Comment:
For the purpose of temporary highlighting, other less intrusive mechanisms can be used (similar to `volatile-highlights`). That can be left as a `TODO` for later, though.

Reply:
So, I'm leaving this as is...